### PR TITLE
[test_snmp_memory_load] reserver 256M free memory

### PR DESCRIPTION
**Pull request by lipxu**
### Description of PR

Summary:
Fixes # (issue)
16303901

### Type of change

- [x] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

### Back port request
- [ ] 201911
- [ ] 202012
- [x] 202205

#### What is the motivation for this PR?
The case failed on 7050qx which is a low memory device

#### How did you do it?
Reserve 256M memory for the system to avoid DUT out of memory

#### How did you verify/test it?
Run original case
snmp/test_snmp_memory.py::test_snmp_memory_load[str2-7050qx-32s-acs-01] PASSED  

#### Any platform specific information?
Low memory device

### CODE REVIEW

File: tests/snmp/memory.py [hunk 1]
```diff
@@ -11,28 +11,35 @@ def get_free_memory():
...
-            chunk = ' ' * min(chunk_size, remaining_chars)
-            large_string += chunk
-            remaining_chars -= chunk_size
-
-        for i in range(0, len(large_string), chunk_size):
-            print(large_string[i:i+chunk_size])
-    except MemoryError:
-        print("Not enough memory to generate and print the large string.")
+    # for small memory device, use chunk size instead of total_chars
+    command = "sudo sh -c 'echo 3 > /proc/sys/vm/drop_caches'"
```
Comment (SuvarnaMeenakshi) [inline tests/snmp/memory.py:28]: if we are reserving the required memory, do we still need to drop_caches?

**Comment by SuvarnaMeenakshi** [comment]
@lipxu upon manual test run, was it tested to ensure that there was a low memory on the device and the memory.py entered the low memory condition inside else block?
I am asking so because in the manual test earlier, I had observed that it never entered the low memory condition.

File: tests/snmp/memory.py (hunk 1 above)
Comment (lipxu) [reply tests/snmp/memory.py:28]: Hi, @SuvarnaMeenakshi , thanks for reviewing the PR.
Based on the test result, on 7050qx, normally, the available memory is about 600M, but the free memory is less than 200M which is less than the threshold of the reserve free memory. 
So I added this CMD to drop the caches, thanks.

**Comment by lipxu** [comment]
> @lipxu upon manual test run, was it tested to ensure that there was a low memory on the device and the memory.py entered the low memory condition inside else block? I am asking so because in the manual test earlier, I had observed that it never entered the low memory condition.

Yes, @SuvarnaMeenakshi , I verified the case locally and made sure the low-memory code path works. thanks.

**Review by SuvarnaMeenakshi — APPROVED** [review]
lgtm

**Pull request merged**